### PR TITLE
Halyard: Support loading BOM from disk rather than GCS #2390

### PR DESCRIPTION
**Pull request by mracter**
@lwander

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Mon Dec 11 14:43:59 EST 2017
+#Tue Apr 03 13:32:29 SAST 2018
```
Comment (lwander) [inline gradle/wrapper/gradle-wrapper.properties]: I think your build system/IDE made these changes. Can you back these out?

File: gradlew [hunk 2]
```diff
@@ -1,17 +1,35 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (lwander) [inline gradlew]: I think your build system/IDE made these changes. Can you back these out?

File: gradlew.bat [hunk 3]
```diff
@@ -8,14 +8,14 @@
 @rem Set local scope for the variables with windows NT shell
 if "%OS%"=="Windows_NT" setlocal
 
-@rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
```
Comment (lwander) [inline gradlew.bat]: I think your build system/IDE made these changes. Can you back these out?

File: halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    String bomPath(String version) {
+      return Paths.get(localBomPath, String.join("/", "bom", version + ".yml")).toString();
+    }
+
+    private InputStream getContents(String objectName) throws IOException {
+        log.info("Getting file contents of " + objectName);
+        return new FileInputStream(objectName);
+      }
+
+    }
```
Comment (lwander) [inline halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java]: This should be unindented

File: halyard-core/src/test/groovy/com/netflix/spinnaker/halyard/core/resource/v1/LocalDiskProfileReaderSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017 Google, Inc.
```
Comment (lwander) [inline halyard-core/src/test/groovy/com/netflix/spinnaker/halyard/core/resource/v1/LocalDiskProfileReaderSpec.groovy]: You should make this 2018, and your company or name

File: halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 Google, Inc.
```
Comment (lwander) [inline halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java]: Same comment here about updating this.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java [hunk 7]
```diff
@@ -67,7 +72,7 @@ String gitRoot(@Value("${git.root:~/dev/spinnaker}") String gitRoot) {
   }
 
   @Bean
-  String spinnakerStagingDependencyPath(@Value("${spinnaker.config.staging.directory:~/.halyard}") String path) {
+  String spinnakerStagingDependencyPath(@Value("${spinnaker.config.staging.directory:~/.hal}") String path) {
```
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java]: Is this fixing an outstanding bug, or a change for this feature?

File: halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java (hunk 4 above)
Comment (mracter) [reply halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java]: Fixed tabstops.

File: halyard-core/src/test/groovy/com/netflix/spinnaker/halyard/core/resource/v1/LocalDiskProfileReaderSpec.groovy (hunk 5 above)
Comment (mracter) [reply halyard-core/src/test/groovy/com/netflix/spinnaker/halyard/core/resource/v1/LocalDiskProfileReaderSpec.groovy]: Changed

File: halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java (hunk 6 above)
Comment (mracter) [reply halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/LocalDiskProfileReader.java]: Changed

**Comment by Tharun-Sabbu** [comment]
Can you please provide the documentation on how to achieve the offline installation of spinnaker.

Thanks

**Comment by rahbal** [comment]
@Tharun-Sabbu still looking for it. Do you have a link for me after 2 years ?

**Pull request merged**